src/erlcloud_ddb2.erl

```diff
@@ -2452,6 +2842,132 @@ q(Table, KeyConditionsOrExpression, Opts, Config) ->
     out(Return, fun(Json, UOpts) -> undynamize_record(q_record(), Json, UOpts) end, DdbOpts, 
         #ddb2_q.items, {ok, []}).
 
+%%------------------------------------------------------------------------------
+%% @doc
+%% DynamoDB API:
+%% [https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_RestoreTableFromBackup.html]
+%%
+%% ===Example===
+%%
+%% Creates a new table from an existing backup.
+%%
+%% `
+%% {ok, Table} =
+%%     erlcloud_ddb2:restore_table_from_backup(<<"BackupArn">>,<<"Forum">>),
+%% '
+%% @end
+%%-------------------------------------------------------------------------------
+-type restore_table_from_backup_return() :: ddb_return(#ddb2_restore_table_from_backup{}, #ddb2_table_description{}).
+
+-spec table_description_with_restore_summary_record() -> record_desc().
+table_description_with_restore_summary_record() ->
```

Comment:
I don't think this is necessary. **All** table descriptions should have restore summary. Update the table_description_record to return it.

See the brokenness below. `restore_summary` is undefined, but it's there in the raw response.

```erlang
16> erlcloud_ddb2:describe_table(<<"nlundgaard.test.restored">>).
{ok,#ddb2_table_description{
        attribute_definitions = [{<<"id">>,s}],
        creation_date_time = 1525874024.54,
        global_secondary_indexes = undefined,item_count = 0,
        key_schema = <<"id">>,latest_stream_arn = undefined,
        latest_stream_label = undefined,
        local_secondary_indexes = undefined,
        provisioned_throughput =
            #ddb2_provisioned_throughput_description{
                last_decrease_date_time = undefined,
                last_increase_date_time = undefined,
                number_of_decreases_today = 0,read_capacity_units = 2,
                write_capacity_units = 2},
        restore_summary = undefined,
        sse_description = {status,enabled},
        stream_specification = undefined,
        table_arn =
            <<"arn:aws:dynamodb:us-east-1:025778144193:table/nlundg"...>>,
        table_name = <<"nlundgaard.test.restored">>,
        table_size_bytes = 0,table_status = creating}}
17> erlcloud_ddb2:describe_table(<<"nlundgaard.test.restored">>, [{out, json}]).
{ok,[{<<"Table">>,
      [{<<"AttributeDefinitions">>,
        [[{<<"AttributeName">>,<<"id">>},
          {<<"AttributeType">>,<<"S">>}]]},
       {<<"CreationDateTime">>,1525874024.54},
       {<<"ItemCount">>,0},
       {<<"KeySchema">>,
        [[{<<"AttributeName">>,<<"id">>},
          {<<"KeyType">>,<<"HASH">>}]]},
       {<<"ProvisionedThroughput">>,
        [{<<"NumberOfDecreasesToday">>,0},
         {<<"ReadCapacityUnits">>,2},
         {<<"WriteCapacityUnits">>,2}]},
       {<<"RestoreSummary">>,
        [{<<"RestoreDateTime">>,1525873784.906},
         {<<"RestoreInProgress">>,true},
         {<<"SourceBackupArn">>,
          <<"arn:aws:dynamodb:us-east-1:025778144193:tabl"...>>},
         {<<"SourceTableArn">>,
          <<"arn:aws:dynamodb:us-east-1:025778144193:"...>>}]},
       {<<"SSEDescription">>,[{<<"Status">>,<<"ENABLED">>}]},
       {<<"TableArn">>,
        <<"arn:aws:dynamodb:us-east-1:025778144193:table/nlundgaard"...>>},
       {<<"TableId">>,<<"6be064b6-7250-46ae-bc4f-5d95d21e7513">>},
       {<<"TableName">>,<<"nlundgaard.test.restored">>},
       {<<"TableSizeBytes">>,0},
       {<<"TableStatus">>,<<"CREATING">>}]}]}
```

Reply:
To clarify a little more, there is no `TableDescriptionWithRestoreSummary` record in DynamoDB. The `RestoreTableFromBackup` operation docs specifically say that the response is just a [`TableDescription`](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_RestoreTableFromBackup.html#DDB-RestoreTableFromBackup-response-TableDescription). That's why I am saying you need to just fix the `table_description_record()` function, because this needs to work when I call `DescribeTable` too.